doc/dev/custom_devices.rst

```diff
@@ -17,91 +17,116 @@ Additionally, all measurements will always return ``true``.
-                                       const std::vector<QubitIdType> &, size_t)
+            auto Expval(ObsIdType) -> double override { return 0.0; }
+            auto Var(ObsIdType) -> double override { return 0.0; }
+            void State(DataView<std::complex<double>, 1> &) override {}
+            void Probs(DataView<double, 1> &) override {}
+            void PartialProbs(DataView<double, 1> &, const std::vector<QubitIdType> &) override {}
+            void Sample(DataView<double, 2> &, size_t) override {}
+            void PartialSample(DataView<double, 2> &, const std::vector<QubitIdType> &, size_t) override {}
+            void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) override {}
+
+            void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
+                                    const std::vector<QubitIdType> &, size_t) override
             {
             }
 
-            virtual void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) {}
+            void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) override {}
         };
 
-In addition to implementing the ``QuantumDevice`` class, one must implement the following method:
+In addition to implementing the ``QuantumDevice`` class, one must implement the following method
+with the name ``CustomDevice`` + ``Factory``:
 
 .. code-block:: c++
 
     extern "C" Catalyst::Runtime::QuantumDevice*
-    getCustomDevice() { return new CustomDevice(); }
+    CustomDeviceFactory(bool status, const std::string &kwargs)
+    {
+        return new CustomDevice(status, kwargs); }
+
+The use of the term ``Factory`` in ``CustomDeviceFactory`` is a runtime convention that combines the name
+of the device class, in this case, ``CustomDevice``, with the suffix ``Factory``. This signifies that the
+primary purpose of the method is to act as a factory for creating instances of the ``CustomDevice`` class.
+
+.. warning::
+
+    The runtime exclusively supports factory methods with the name of the device class followed by ``Factory``.
+    Please ensure that your method adheres to this naming convention.
 
-where ``CustomDevice()`` is a constructor for your custom device.
-``CustomDevice``'s destructor will be called by the runtime.
+``CustomDevice(status, kwargs)`` serves as a constructor for your custom device, with
 
-.. note::
+- ``status``: A boolean to control the initial value of the tape recording system in the runtime.
+  If ``true``, this will cache the entire operations and observables in a program.
+  This may be used to compute gradient of a quantum circuit with taking advantage of gradient
+  methods provided by backend devices if supported.
+
+- ``kwargs``: A string that represents a Python dictionary of device specifications in string format.
+  This may contain information about the device type and a number of shots:
+  ``{'device_type': 'lightning.qubit', 'shots': 0}``.
```

Comment:
```suggestion
- ``kwargs``: A string of device specifications and options, represented in Python dictionary format.
  This may contain information such as device type and a number of shots:
  ``{'device_type': 'lightning.qubit', 'shots': 0}``.
```

Reply:
What is the purpose of `device_type` here?

Reply:
I would like to preserve `device_type`. I at least like it for the toml files. Right now the names are `lightning.qubit.toml` and that is f`{device_type}.toml`.

Reply:
If `device_type` is a holdover from before, or specific to only one device, we might want to use a more general example of a parameter a device might accept. But the important part is that this is not a specification, and is just showing that devices can receive arbitrary information from the program in string format.

Reply:
This is technically needed for `openqasm` devices to declare the type of the device (remote or local). I re-scoped this to be only appeared for `openqasm` devices (and not for lightning simulators).